### ISSUE TITLE
Not possible to export scene with linked objects

### ISSUE THREAD

**Issue by slapin** [godotengine/godot-blender-exporter#247]
Error:
```
exporting /home/slapin/godot-projects/kicking-high/proto2/assets/blender-2.80/level.blend
Read blend: /home/slapin/godot-projects/kicking-high/proto2/assets/blender-2.80/./level.blend
Info: Read library:  '/home/slapin/godot-projects/kicking-high/proto2/assets/blender-2.80/level-kit.blend', '//level-kit.blend', parent '<direct>'
Info: Read library:  '/home/slapin/godot-projects/kicking-high/proto2/assets/blender-2.80/level-kit.blend', '//level-kit.blend', parent '<direct>'
[INFO]: Found Godot project directory at /home/slapin/godot-projects/kicking-high/proto2/./.
[INFO]: Exporting scene: Scene
[INFO]: Exporting 116 objects
[INFO]: Exporting Blender object: room_owall_anglesw1_3-col.001
[INFO]: Exporting Blender object: room_owall1_3-col.001
Traceback (most recent call last):
  File "/home/slapin/godot-projects/kicking-high/proto2/export.py", line 115, in run_with_abort
    function()
  File "/home/slapin/godot-projects/kicking-high/proto2/export.py", line 95, in main
    export_escn(out_path, config)
  File "/home/slapin/godot-projects/kicking-high/proto2/export.py", line 13, in export_escn
    io_scene_godot.export(out_file, config)
  File "/home/slapin/blender-2.80-linux-glibc217-x86_64/2.80/scripts/addons/io_scene_godot/__init__.py", line 256, in export
    export_godot.save(FakeOp(), bpy.context, filename, **default_settings)
  File "/home/slapin/blender-2.80-linux-glibc217-x86_64/2.80/scripts/addons/io_scene_godot/export_godot.py", line 318, in save
    exp.export()
  File "/home/slapin/blender-2.80-linux-glibc217-x86_64/2.80/scripts/addons/io_scene_godot/export_godot.py", line 267, in export
    self.export_scene()
  File "/home/slapin/blender-2.80-linux-glibc217-x86_64/2.80/scripts/addons/io_scene_godot/export_godot.py", line 211, in export_scene
    self.export_object(obj, root_gd_node)
  File "/home/slapin/blender-2.80-linux-glibc217-x86_64/2.80/scripts/addons/io_scene_godot/export_godot.py", line 123, in export_object
    parent_gd_node)
  File "/home/slapin/blender-2.80-linux-glibc217-x86_64/2.80/scripts/addons/io_scene_godot/converters/mesh.py", line 45, in export_mesh_node
    mesh_id = mesh_exporter.export_mesh(escn_file, export_settings)
  File "/home/slapin/blender-2.80-linux-glibc217-x86_64/2.80/scripts/addons/io_scene_godot/converters/mesh.py", line 159, in export_mesh
    key = MeshResourceKey('ArrayMesh', self.object, export_settings)
  File "/home/slapin/blender-2.80-linux-glibc217-x86_64/2.80/scripts/addons/io_scene_godot/converters/utils.py", line 81, in __init__
    self._hash = hash(self._data)
TypeError: Vector is not frozen (mutable), call freeze first
```

**Comment by k0nfus1us** [comment]
A bit of debugging:
The function at line #123 of export_godot.py takes an `obj` parameter, printing the `obj.type` shows the error occurs for `MESH` types.

**Comment by dam482** [comment]
This same error can be produced by using the default blender scene, and adding an Array modifier onto the cube.  Then export the scene using the default settings.  Don't think it has anything specific to do with linked objects.

**Comment by slapin** [comment]
Somehow linking triggers the same problem too on my side.

**Comment by dam482** [comment]
According to the error, the object "room_owall1_3-col.001" is the problem.  Does it have an array modifier on it?

**Comment by slapin** [comment]
No, but it has shared data (Alt-D) if that is important.

**Comment by slapin** [comment]
To explain - it was linked from another file and Alt-D'ed.

**Comment by hartsantler** [comment]
i'm having the same problem, but i think its linked materials that cause my error:
```
[INFO]: Exporting scene: Scene
[INFO]: Exporting 211 objects
[INFO]: Exporting Blender object: Cube
[INFO]: Exporting Blender object: Light
[INFO]: Exporting Blender object: Camera
[INFO]: Exporting Blender object: __topsoil__.001
Array iterator out of range: Mesh_materials_lookup_int (index 16)
Error: Traceback (most recent call last):
  File "/home/raptor/Downloads/blender-2.80/2.80/scripts/addons/io_scene_godot/__init__.py", line 198, in execute
    return export_godot.save(self, context, **keywords)
  File "/home/raptor/Downloads/blender-2.80/2.80/scripts/addons/io_scene_godot/export_godot.py", line 318, in save
    exp.export()
  File "/home/raptor/Downloads/blender-2.80/2.80/scripts/addons/io_scene_godot/export_godot.py", line 267, in export
    self.export_scene()
  File "/home/raptor/Downloads/blender-2.80/2.80/scripts/addons/io_scene_godot/export_godot.py", line 211, in export_scene
    self.export_object(obj, root_gd_node)
  File "/home/raptor/Downloads/blender-2.80/2.80/scripts/addons/io_scene_godot/export_godot.py", line 123, in export_object
    parent_gd_node)
  File "/home/raptor/Downloads/blender-2.80/2.80/scripts/addons/io_scene_godot/converters/mesh.py", line 45, in export_mesh_node
    mesh_id = mesh_exporter.export_mesh(escn_file, export_settings)
  File "/home/raptor/Downloads/blender-2.80/2.80/scripts/addons/io_scene_godot/converters/mesh.py", line 175, in export_mesh
    mesh
  File "/home/raptor/Downloads/blender-2.80/2.80/scripts/addons/io_scene_godot/converters/mesh.py", line 281, in generate_surfaces
    mat = mesh.materials[face.material_index]
IndexError: bpy_prop_collection[index]: index 16 out of range, size 16

location: /home/raptor/Downloads/blender-2.80/2.80/scripts/modules/bpy/ops.py:201

```

**Comment by Jason0214** [comment]
Looks like the new hashing is causing issue, there are some property of the blender object which is not directly hashable.


> it has shared data (Alt-D) if that is important.
@slapin  linked data should be fine, we have test coverage of it, but still thanks for providing the information


@hartsantler thanks for the testing! But looks like you find a different issue than the original one😅

**Comment by Jason0214** [comment]
Should be fixed by #262
